opensearch-operator/controllers/clusterController.go

```diff
@@ -4,122 +4,78 @@ import (
 	"context"
 	"fmt"
 
-	sts "k8s.io/api/apps/v1"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/record"
 	opsterv1 "opensearch.opster.io/api/v1"
 	"opensearch.opster.io/pkg/builders"
-	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type State struct {
-	Component string `json:"component,omitempty"`
-	Status    string `json:"status,omitempty"`
-	Err       error  `json:"err,omitempty"`
-}
-
 type ClusterReconciler struct {
 	client.Client
-	Scheme   *runtime.Scheme
 	Recorder record.EventRecorder
-	State    State
+	logr.Logger
 	Instance *opsterv1.OpenSearchCluster
 }
 
-//+kubebuilder:rbac:groups="opensearch.opster.io",resources=events,verbs=create;patch
-//+kubebuilder:rbac:groups=opensearch.opster.io,resources=opensearchcluster,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=opensearch.opster.io,resources=opensearchcluster/status/componentsStatus,verbs=get;update;patch
-//+kubebuilder:rbac:groups=opensearch.opster.io,resources=opensearchcluster/finalizers,verbs=update
+func (r *ClusterReconciler) Reconcile(controllerContext *ControllerContext) (*opsterv1.ComponentStatus, error) {
 
-func (r *ClusterReconciler) Reconcile(context.Context, ctrl.Request) (ctrl.Result, error) {
-
-	cm := v1.ConfigMap{}
 	namespace := r.Instance.Spec.General.ClusterName
 
-	cmName := "opensearch-yml"
```

Comment:
didnt understand why you deleted the configmap creation phase

Reply:
I moved this into the configuration controller. It creates the configmap and instructs the cluster controller via the ControllerContext to mount it.